bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -82,87 +85,90 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                                        CHANNEL_LOADPOINT_TARGET_TIME);
-                                updateState(channel, new DateTimeType(targetTimeZDT));
-                                if (targetTimeEnabled) {
-                                    try {
-                                        evccAPI.setTargetTime(loadpoint, targetTimeZDT);
-                                    } catch (DateTimeParseException e) {
-                                        logger.debug("Failed to set target charge: ", e);
-                                    }
-                                }
+                        case CHANNEL_RESIDUAL_POWER -> {
+                            if (command instanceof QuantityType<?> qt) {
+                                evccAPI.setResidualPower(qt.toUnit(Units.WATT).intValue());
+                            } else if (command instanceof DecimalType dt) {
+                                evccAPI.setResidualPower(dt.intValue());
                             } else {
-                                logger.debug("Command has wrong type, DateTimeType required!");
+                                logger.debug("Command has wrong type, QuantityType or DecimalType required!");
                             }
                         }
-                        case CHANNEL_LOADPOINT_TARGET_TIME_ENABLED -> {
+                        case CHANNEL_BATTERY_DISCHARGE_CONTROL -> {
                             if (command == OnOffType.ON) {
-                                evccAPI.setTargetTime(loadpoint, targetTimeZDT);
-                                targetTimeEnabled = true;
+                                evccAPI.setBatteryDischargeControl(true);
                             } else if (command == OnOffType.OFF) {
-                                evccAPI.removeTargetTime(loadpoint);
-                                targetTimeEnabled = false;
+                                evccAPI.setBatteryDischargeControl(false);
                             } else {
                                 logger.debug("Command has wrong type, OnOffType required!");
                             }
                         }
+                        default -> {
+                            return;
+                        }
+                    }
+                } else if (channelGroupId.startsWith(CHANNEL_GROUP_ID_LOADPOINT)) {
+                    int loadpoint = Integer.parseInt(groupId.substring(9)) + 1;
+                    switch (channelIdWithoutGroup) {
+                        case CHANNEL_LOADPOINT_MODE -> {
+                            if (command instanceof StringType) {
+                                evccAPI.setMode(loadpoint, command.toString());
+                            } else {
+                                logger.debug("Command has wrong type, StringType required!");
+                            }
+                        }
+                        case CHANNEL_LOADPOINT_LIMIT_ENERGY -> {
```

Comment:
Please add support for DecimalType commands, interpret them as kWh.

Reply:
ack